sdk/core/System.ClientModel/src/Options/ClientLoggingOptions.cs

```diff
@@ -0,0 +1,283 @@
+    /// typically hold sensitive information.</value>
+    public IList<string> AllowedHeaderNames
+    {
+        get
+        {
+            if (!_frozen)
+            {
+                if (_allowedHeaderNames is null)
+                {
+                    var changeList = new ChangeTrackingStringList(s_defaultAllowedHeaderNames);
+                    changeList.StartTracking();
+                    _allowedHeaderNames = changeList;
+                }
+                return _allowedHeaderNames;
+            }
+            else
+            {
+                if (_allowedHeaderNames is null)
+                {
+                    // If this instance is frozen still allow read-only access to the defaults by
+                    // creating a copy of the default allowed headers and freezing it. This
+                    // avoids copying the default array and allocating the change tracking list unless necessary.
+                    _allowedHeaderNames = new ChangeTrackingStringList(s_defaultAllowedHeaderNames);
+                    _allowedHeaderNames.Freeze();
+                }
+                return _allowedHeaderNames;
+            }
+        }
+    }
+
+    /// <summary>
+    /// Gets or sets a list of query parameter names that are not redacted during logging.
+    /// </summary>
+    /// <value>Defaults to a list of common query parameters that do not
+    /// typically hold sensitive information.</value>
+    public IList<string> AllowedQueryParameters
+    {
+        get
+        {
+            if (!_frozen)
+            {
+                if (_allowedQueryParameters is null)
+                {
+                    var changeList = new ChangeTrackingStringList(s_defaultAllowedQueryParameters);
+                    changeList.StartTracking();
+                    _allowedQueryParameters = changeList;
+                }
+                return _allowedQueryParameters;
+            }
+            else
+            {
+                if (_allowedQueryParameters is null)
+                {
+                    // If this instance is frozen still allow read-only access to the defaults by
+                    // creating a copy of the default allowed query parameters and freezing it. This
+                    // avoids copying the default array and allocating the change tracking list unless necessary.
+                    _allowedQueryParameters = new ChangeTrackingStringList(s_defaultAllowedQueryParameters);
+                    _allowedQueryParameters.Freeze();
+                }
+                return _allowedQueryParameters;
+            }
+        }
+    }
+
+    /// <summary>
+    /// Freeze this instance of <see cref="ClientLoggingOptions"/>.  After this method
+    /// has been called, any attempt to set properties on the instance or call
+    /// methods that would change its state will throw <see cref="InvalidOperationException"/>.
+    /// </summary>
+    public virtual void Freeze()
```

Comment:
why is this virtual?

Reply:
This is following the pattern on ClientPipelineOptions: https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/core/System.ClientModel/src/Options/ClientPipelineOptions.cs#L164